### PR TITLE
refactor: enhance button component

### PR DESCRIPTION
**Pull request by tinaClin**
## Change List
- Limit `variant` property to specific values `emphasis`, `primary`, `default`, `secondary` and `ghost`
- Remove `variantColor` property from Button
- Renaming Button's property that selected state from `active` to `selected`

## Button
image

## ButtonGroup
image

### CODE REVIEW

File: packages/styled-docs/pages/buttongroup.mdx [hunk 1]
```diff
@@ -133,7 +121,7 @@ Make a set of buttons appear vertically stacked rather than horizontally, by add
 
 | Name | Type | Default | Description |
 | :--- | :--- | :------ | :---------- |
-| `size` | `sm`, `md`, `lg` | `md` | Sets the size for all Buttons in the group. |
-| `variant` | `solid`, `outline` | `solid` | Sets the variant of all Buttons in the group. |
-| `variantColor` | string | | Sets the variant of all Buttons in the group. Use a color key passed in `theme.colors`. |
-| `vertical` | boolean | `false` | Make the set of Buttons appear vertically stacked. |
+| `nodivide` | `boolean` | false | There are no the divide line between buttons. |
```
Comment (cheton) [inline packages/styled-docs/pages/buttongroup.mdx]: 這個 prop name 我看不下去 XD

Comment (cheton) [reply packages/styled-docs/pages/buttongroup.mdx]: I can't imagine why this prop is really needed.

Comment (tinaClin) [reply packages/styled-docs/pages/buttongroup.mdx]: `emphasis`, `primary`, `default` 這三種button需要中間的分隔線
`secondary`,`ghost`這兩種button不需要

如果不用nodivide去判斷需不需要加分隔線，那就只好判斷variant?

File: packages/react-styled-core/src/Button/index.js [hunk 2]
```diff
@@ -1,39 +1,60 @@
...
+    }
 
     return (
       <ButtonBase
         ref={ref}
         as={Comp}
         type={type}
         borderRadius={borderRadius}
-        data-active={active ? 'true' : undefined}
+        data-selected={selected ? 'true' : undefined}
```
Comment (cheton) [inline packages/react-styled-core/src/Button/index.js:41]: `PseudoBox` 是用 `data-active` 屬性

```js
const selected = [
  '&[data-active=true]',
  '&:active[data-active=true]',
  '&:hover[data-active=true]',
];
```

File: packages/styled-docs/pages/buttongroup.mdx (hunk 1 above)
Comment (cheton) [reply packages/styled-docs/pages/buttongroup.mdx]: 對呀! 我的意思就是說靠 variant 就可以了

File: packages/react-styled-core/src/Button/index.js (hunk 2 above)
Comment (tinaClin) [reply packages/react-styled-core/src/Button/index.js:41]: `PseudoBox`那邊是我加的，我也有一起改了~

Comment (cheton) [reply packages/react-styled-core/src/Button/index.js:41]: https://github.com/trendmicro-frontend/styled-ui/blob/3da7e1bdd2477e4c42eccfdb1765c25f71d1e44e/packages/react-styled-core/src/PseudoBox/index.js#L29-L33

Comment (cheton) [reply packages/react-styled-core/src/Button/index.js:41]: ok!

**Review by cheton — APPROVED** [review]

File: packages/styled-docs/pages/buttongroup.mdx (hunk 1 above)
Comment (tinaClin) [reply packages/styled-docs/pages/buttongroup.mdx]: commit [f5d1750].

**Review by roth1002 — APPROVED** [review]

**Review by austinhung81 — APPROVED** [review]

**Pull request merged**